### PR TITLE
[ACHIEVEMENT] Meep Meep

### LINKED ISSUE #191: [ACHIEVEMENT] Meep Meep
###### ACHIEVEMENT
---
Achievement for commenting on a pull request within a 5 minutes period

## Reference
We all know [The Road Runner](https://youtu.be/oo3aC46180s)

## Metadata (suggestion) :

name: Meep Meep

image: https://media.giphy.com/media/xThtaaVNrPdQ64U9vq/giphy.gif (made to a square)

short: Wile E. Coyote will stay hungry today 

description:
`You've commented on a Pull Request within 5 minutes - that pretty quick... for a human`

### PR DESCRIPTION
**Pull request by ZimGil**
# Added the "Meep Meep" achievement

 - New Achievement: "Meep Meep" [resolves #191]
    > Will grant an achievement to the author of the first comment within 5 minutes of the Pull Request creation.
    > Will work with either comment and inline comments, but not both - only the first

### CODE REVIEW

**Comment by ortichon** [comment]
Kinda harsh requirements:
✖ message may not be empty
✖ type may not be empty

**Comment by ZimGil** [comment]
This is actually require changes.
There's need to be a test that will make sure that `creator` will not get the achievement,
the achievement should be modified accordingly.

**Comment by ortichon** [comment]
@ZimGil @Thatkookooguy 
We should address the issue of bots/hound.
do their comments count as comments?
if so, we should filter them out. and actually, update this achievement for each new bot we add?

**Comment by Thatkookooguy** [comment]
> @ZimGil @Thatkookooguy
> We should address the issue of bots/hound.
> do their comments count as comments?
> if so, we should filter them out. and actually, update this achievement for each new bot we add?

I'm not sure if we mind bots getting achievements. Just like the greenkeeper bot can get achievements. We don't discriminate bots lol

I'm trying to think if 5 minutes is not too long for just a "first comment" and not a "first review comment". review comments should take time, but regular comments are easy to add and can easily be shorter than 5 minutes

**Comment by ortichon** [comment]
I don't have a problem with bots getting achievements too, but then @ZimGil will have to change his logic.
IIRC the current logic gives the achievement to the first (under 5 minutes) commenter. 
I guess we'll have to change to logic to give the achievement to ALL under 5 minutes commenters?

**Comment by Thatkookooguy** [comment]
> I don't have a problem with bots getting achievements too, but then @ZimGil will have to change his logic.
> IIRC the current logic gives the achievement to the first (under 5 minutes) commenter.
> I guess we'll have to change to logic to give the achievement to ALL under 5 minutes commenters?

I think it just should be shorter. Most bots write a comment based on something that happened.

If the PR format or commit format is wrong, or for a first PR (the welcome bot), the comment will be fast. But for lint errors, tests, coverage, etc., bot comments should be "not as fast".

So we need to test this a bit I guess but it sounds to me like in `ver2`, the bots that will get the achievements are the **welcome-bot**, and **commitlint-bot**. All the other bots won't run fast enough.

Base on what @ZimGil wants, I think we should do one of the two:
- make this achievement only for "review comments". That's basically a human domain, so if we want to isolate this to humans, that makes sense. Also, a review is hard to do under 5 minutes unless the PR is very small. So I think that makes sense

- shorten the 5 minutes to something more challenging. writing a **normal comment** takes nothing. 5 minutes sounds a bit slow IMO for a Road Runner reference :-)
  for normal comments I'll go for something like 2 minutes? what do you guys think?

so basically, that depends on what we, and especially @ZimGil , thinks will be more fun\funny when getting this achievement

**Comment by Thatkookooguy** [comment]
We should maybe test this on a repo with bots installed and see if everything works as expected

**Comment by ZimGil** [comment]
> Base on what @ZimGil wants, I think we should do one of the two:
> 
> * make this achievement only for "review comments". That's basically a human domain, so if we want to isolate this to humans, that makes sense. Also, a review is hard to do under 5 minutes unless the PR is very small. So I think that makes sense
> * shorten the 5 minutes to something more challenging. writing a **normal comment** takes nothing. 5 minutes sounds a bit slow IMO for a Road Runner reference :-)
>   for normal comments I'll go for something like 1-2 minutes? what do you guys think?
> 
> so basically, that depends on what we, and especially @ZimGil , thinks will be more fun\funny when getting this achievement

I actually think both case fit the reference and are funny.
With that said, I do think **review comments** are more eligible for this achievement.
